Use 0x for CIO.

hrvdm opened the pull request.

hrvdm committed:
use 0x as ID

elffjs commented on internal/controller/controller.go:
```diff
@@ -213,7 +213,7 @@ func (d *Controller) createUser(ctx context.Context, userAccount *AccountClaims,
 		}
 
-		if err := d.cioService.SetEmail(ctx, acct.ID, normalEmail); err != nil {
+		if err := d.cioService.SetEmail(ctx, acct.ID, *userAccount.EthereumAddress, normalEmail); err != nil {
```
We're creating the user, and we only have email, so`userAccount.EthereumAddress` is almost certainly nil and the handler will panic.

elffjs replied:
I think if you could guarantee that you'll never create an account with email _first_ then we might be able to make this work. In fact, we would probably want to think about getting rid of this `id` KSUID altogether. Is this the case?

elffjs replied:
Okay, I'm looking at [these lines](https://github.com/DIMO-Network/accounts-api/blob/cca3027ab2c71694e74f74de1b05cc27cd7a9443/internal/controller/crud_handlers.go#L51-L53). In [the logs](https://kibana.team.dimo.zone/app/r/s/rgvM9) it seems like we always start with wallet. I have no idea why the other path is there, then.

elffjs commented on internal/services/cio/cio.go:
```diff
@@ -64,7 +64,7 @@ func (c *Client) SetWallet(ctx context.Context, id string, wallet common.Address
 	}
 
 	return errors.Join(err, c.client.Enqueue(analytics.Identify{
-		UserId: id,
```
Any reason to pass `id` in?

elffjs commented on internal/services/cio/cio.go:
```diff
@@ -38,7 +38,7 @@ func New(settings *config.Settings, logger *zerolog.Logger) (*Client, error) {
 }
 
-func (c *Client) SetEmail(ctx context.Context, id, email string) error {
+func (c *Client) SetEmail(ctx context.Context, id string, wallet common.Address, email string) error {
```
Again, does `id` do anything now?

elffjs commented on internal/controller/email_handlers.go:
```diff
@@ -95,7 +95,7 @@ func (d *Controller) LinkEmail(c *fiber.Ctx) error {
 	logger.Info().Msgf("Added unconfirmed email %s to account.", normalAddr)
 
-	if err := d.cioService.SetEmail(c.Context(), acct.ID, normalAddr); err != nil {
+	if err := d.cioService.SetEmail(c.Context(), acct.ID, *userAccount.EthereumAddress, normalAddr); err != nil {
```
This has a better chance of not blowing up.

elffjs committed:
No more KSUID in Customer.io functions

Guard against the (hopefully never-occurring) possibility that
creation happens under an email JWT

elffjs committed:
One more guard

elffjs committed:
Remove Mixpanel dependencies
